rust_src/src/editfns.rs

```diff
@@ -297,6 +298,24 @@ pub fn following_char() -> EmacsInt {
     }
 }
 
+/// Return the character preceding point, as a number. At the
+/// beginning of the buffer or accessible region, return 0.
+#[lisp_fn(c_name = "previous_char")]
+pub fn preceding_char() -> LispObject {
+    let buffer_ref = ThreadState::current_buffer();
+
+    if buffer_ref.pt <= buffer_ref.begv {
+        return LispObject::from_natnum(0);
+    }
+
+    let pos = if LispObject::from_raw(buffer_ref.enable_multibyte_characters).is_not_nil() {
```

Comment:
instead of making a LispObject just for the comparison you can say `if buffer_ref.enable_multibyte_characters != Qnil`